docs/syntax.md

```diff
@@ -0,0 +1,255 @@
+# LambdaBuffers file
+
+The input to LambdaBuffers is a text file which contains a module that defines
+    a specification of the types you want to generate.
+This section gives the exact syntax of a LambdaBuffers file, and informally describes meaning of the syntactic constructs.
+
+The name of a LambdaBuffers file must end with `.lbf`.
+
+## Notation
+In the following description of a LambdaBuffers file's syntax, we use
+    a similar BNF syntax from [Section 10.1 of the Haskell Report](https://www.haskell.org/onlinereport/haskell2010/).
+So, the following notational conventions are used for presenting syntax.
```

Comment:
How did this pass the markdown checker? Add whitespace after title and fix the whitespace everywhere